extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -418,6 +487,21 @@ export class AmpStoryEmbeddedComponent {
     return parseUrlDeprecated(elUrl).href;
   }
 
+  /**
+   * Gets corresponding config for a given embed target.
+   * @param {!Element} target
+   * @return {!Object}
+   */
+  getEmbedConfigFor_(target) {
+    const componentIndex = findIndex(Object.values(INTERACTIVE_COMPONENTS),
+        config => {
+          return matches(target, config.selector);
+        });
+
+    return INTERACTIVE_COMPONENTS[
+        Object.keys(INTERACTIVE_COMPONENTS)[componentIndex]];
+  }
```

Comment:
At this point, it becomes pretty hard to know how the config is used (Object.values / Object.keys / ...). Object.keys and Object.values loop on the array, findIndex too, but it's mostly becoming hard to follow.
You have much more context on all this, but do you think we could avoid all these by using the `target.tagName` as keys of the config?
I was hoping that this could remove the need for the `getComponentSelectors` function, and would make this method as easy as something like `return COMPONENTS[target.tagName.toLowerCase()];`

Reply:
As chatted offline: We want to support both selectors and tag names for our config, which introduces some overhead at lookup. 

But I agree we are able to cut some complexity by using tag names in the keys and still check for their selectors once we get their config. 
Re. `getComponentSelectors` we unfortunately still need it to export the list of selectors to `page-advancement` so that it knows that it should delegate to this class. (On the upside, it only runs once).

Reply:
Wow that's so much easier, thanks for improving this!